File: modules/database/models/vote.py
from modules.database.basemodel import *
from .server import Server

# This model is created website-site so we won't create it
# from our pov, do it if you need
class Vote(BaseModel):
    id = BigIntegerField(unique=True)
    username = CharField()
    server_id = BigIntegerField()
    created_at = TimestampField()
    updated_at = TimestampField()

    class Meta:
        table_name = 'votes'

def get_all_votes_count():
    try:
        return Vote.select().count()
    except DoesNotExist:
        return 0

def get_server_votes_count(server_id: int):
    try:
        return Vote.select().where(Vote.server_id == server_id).count()
    except DoesNotExist:
        return 0


def get_top_voted_servers(amount:int = 3):
    servers = Server.select()

    for server in servers:
        server.votes = get_server_votes_count(server.id)
        
    return sorted(servers, key=lambda x: x.votes, reverse=True)[:amount]

File: src/models/Vote.js
const {DataTypes, Model} = require("sequelize");
const Server = require("./Server");

module.exports = class Vote extends Model {
    static init(sequelize, options) {
        super.init({
            id: {
                type: DataTypes.BIGINT,
                primaryKey: true,
                allowNull: false,
                unique: true,
            },
            username: {
                type: DataTypes.STRING,
                allowNull: false,
            },
            server_id: {
                type: DataTypes.BIGINT,
                allowNull: false,
            },
            created_at: {
                type: DataTypes.DATE,
                allowNull: false,
            },
            updated_at: {
                type: DataTypes.DATE,
                allowNull: false,
            },
        }, {
            sequelize,
            modelName: 'vote',
            tableName: 'votes',
            timestamps: true,
            createdAt: 'created_at',
            updatedAt: 'updated_at',
        });
        Vote.belongsTo(Server, { foreignKey: 'server_id' });
    }
};